### DISCUSSION TITLE
Nozzle Body Solenoids

### DISCUSSION THREAD

**Discussion by Minorplanets76** [geezacoleman/OpenWeedLocator#80] [Ideas]
I was inspired by a Weedseeker unit at a recent clearing sale.  These have Teejet E-Chemsaver Valves which are mounted directly to the nozzle body.  Sadly, these are very expensive, but on the face of it seem fairly simple as to how they work.

The operation of drip free nozzle operations seems consistent across manufacturers.  I have Arag bodies but also have some original Hardi versions which are the same principle.  Basically there is a diaphram with a spring loaded plunger which blocks off the nozzle under a certain pressure.  Basically my idea is to have a way of holding that diapram closed under working pressure, shutting of the nozzle individually.

I ordered some barrel style solenoids and designed up a 3d printed adapter to mount them directly to the nozzle body.  The only modification to the existing components was to drill a hole in the red spring/plunger retainer so the solendoid could press directly on the plunger.  For the 25mm solenoid, I also had to design a little piston as the diameter of the solenoid rod was too big.

Long story short, only the biggest solenoid was able to hold the nozzle shut, and even then it could only keep it closed, not overcome the resistance needed to close it in the first place.  Testing was done from 2-4 bar, and pressing on the plunger by hand was possible to close but needed a fair bit of force.  The solenoids also get very hot.

In other words, the first iteration was a failure.

I have ordered some water solenoid valves as they should be able to handle the pressure, so I'll see how they work.

The intent is to make a solution that doesn't require re-doing the entire boom, is not cost prohibitive or dearer that just buying a Teejet version, does't require special tools such as lathes and milling machines (but I'm pretending a 3D printer isn't a special tool). I would also like to keep PWM on the table.  Power usage will also be a consideration (the big solenoid was pulling more power than the Pi).

Hopefully it's as simple as designing an adapter that straps an existing valve.  I can think of plenty of other force multiplying solutions like servos, or motors, but then the PWM aspect becomes hard.  Looking at the Teejet maintennce manual, they've solved it by getting rid of the diaphram and having a needle-like plunger.

Any suggestions are welcome.  I will keep my progress documented here.  Designs are in Onshape and are public.

**Comment by geezacoleman** [comment]
This would be excellent - nozzles are such a limiting/costly factor for spot-spraying currently. I know @EchelonMirror has worked on solenoids before and might have some insights here.

What material are you using for the 3D printed parts? Longevity/durability with 3D prints does seem like a challenge, but there are some pretty good materials for filament out there these days.

**Reply by Minorplanets76** [reply]
Just using PETG for testing. Haven't thought too much about longevity as these sort of things seem to iterate faster than they wear out. The big solenoid would probably snap off in a rough paddock so it will be important to keep it relatively compact. 

It's turning out to be a bit of a rabbit hole. Turns out there's different versions of the Teejet chemsaver range.

**Reply by EchelonMirror** [reply]
I've made up a handful of solenoid fittings for mostly TeeJet but covers other brands as well. The body needs to be made of 316 stainless because we're dealing with caustic compounds that will quickly degrade something like PET. 

Depending on the solenoids you're using, they will draw upwards of an Amp and they will get warm to the touch if you leave them on all the time. Though the temp shouldn't get above 40°C. In the situation that we're dealing with where they are only on 15% of the time or are running on PWM temp and wattage shouldn't be an issue. 

The solenoids I designed are operating at 12v / 7 watts (0.58amps), 316 stainless steel, internal plunger modelled off a few different designs out there, operating at up to 30hz. 

If you want I can drop the schematics online that you can send off to a 3rd party to get fabricated, or I can send you a sample of some of the ones that I have made that are compatible with TeeJet. There are a few little things that I need to change, like improving the no drip capability at a different range of back pressures. 

The intention of my design was the user only has to replace the existing TeeJet caps with the solenoid and add the appropriate wiring loom direct to the relay/FET driver interface.

**Reply by Minorplanets76** [reply]
Excellent! Exactly where I was heading with it. I was trying to avoid manufacturing but I do have a lathe & milling machine and not a lot of nozzles 😉. If you could share the schematics that would be great!

The main difference between Hardi and ARAG was the cap thread, and I'd bet Teejet is different again. 

Looking at the Teejet catalog I see that the Dynajet seems to be stainless and pwm is running at 20Hz so it seems you've saved me quite a bit of trial and error. 

Have you wound your own coils? The solenoids I tested were chosen for being conveniently round and threaded. They are probably used as a locking mechanism and must have a very low duty cycle. The only one that could work using a diaphragm was something like 25W so impractical.

**Comment by Minorplanets76** [comment]
Here is my failed attempt.  The top one was way too weak, and the plunger is now lost until I find it one day in a punctured tyre.  The bottom one nearly works.  It can hold but not close from an open state.  Plus it's a bit comically long, heavy and power hungry.

![solenoids](https://github.com/geezacoleman/OpenWeedLocator/assets/107326485/b7fd7f01-bf56-44ed-a143-66697572fbc5)

3d printed thread adapter with no wetted parts, and only mod required is drilling a hole in red piece for the ARAG body. The Hardi Body already has the hole.  This way, the diapham still does the work, but you are simply increasing "non-drip" pressure.  The ARAG body has quite a big orifice going to the nozzle, so by my calcs it needs at least 10-15kg to hold, which is why it's struggling.  The Hardi version might work.

Looking at other valves, they all seem to be a pilot operated style, where a wetted chamber does all the work with a pressure differential.  Kind of like check valves that can be opened. The wetted chamber is probably not a good candidate for 3D printing but I guess I could try some more exotic filaments.

The strength of the solenoid then becomes a function of the orifice size. The pressure drop and PWM characteristics will also start interacting with the nozzle size, so you will probably need to tune.

I've checked a couple of regular fluid control solenoids to see if they would be adaptable, choosing ones that seem to act directly rather than the ones with the big rubber flap with a stainless bit in the middle.

![solenoids2](https://github.com/geezacoleman/OpenWeedLocator/assets/107326485/00f4e7a7-5a66-4ff7-b40c-651321f59b53)

One seems ideal.  It already has a good plunger and is a good size.  It would just need a chamber adapted to it.  When I went to go buy a dozen more, they suddenly seem to be sold out. The other, far more common one uses a much simpler plunger.  The problem with it is the solenoid is so big, it won't fit on the Hardi body, and would likely interfere with end nozzles on an ARAG wet boom as well.  The small one can apparently do 70bar so only has a 0.8mm orifice, whereas the big one has an ~2mm orifice. I might have a look at the internals of my current section control valves next.  I get them from [https://valvesdirect.net/](url) with the chemical resistant option and they seem pretty good value.

**Reply by EchelonMirror** [reply]
Another factor to consider for the average install is solenoid costs. If you're forking out $300 per solenoid on 25cm spacing on a 36m boom you're bill is over $43,000. Yes it adds value, but it can add a sizeable cost to an install. I'm currently moving to a design (iteration 2 if you will) that looks similar to that Yong Chuang one you have there. Has a similar mounting to the Tee Jet.

**Reply by Minorplanets76** [reply]
Yes, they are expensive alright. Of course, at the rate I'm going, I'll end up spending more trying to prove I can do it cheaper.

The valvesdirect valves won't work as they're a bit big. They have quite a small plunger with a  viton cylinder on the end.  Rest is plastic with what looks like nylon with a rubber gasket cast around it.

I haven't given up on my original concept, but have given up on that being good enough for pwm. Question becomes are they quick enough for Owl.  I'm fiddling with a cam or screw type blocking mechanism at the moment, but keeping the part count and complexity low is tricky.

**Reply by Minorplanets76** [reply]
Do you have a link to the solenoid you're using?
